pkg/prometheus/promcfg.go

```diff
@@ -348,6 +381,34 @@ func addTLStoYaml(cfg yaml.MapSlice, namespace string, tls *monitoringv1.TLSConf
 	return cfg
 }
 
+func mergeTLSConfigAndScrapeClassTLSConfig(tlsConfig *monitoringv1.TLSConfig, scrapeClass *monitoringv1.ScrapeClass) *monitoringv1.TLSConfig {
```

Comment:
(suggestion) it could be a ConfigGenerator method:
```suggestion
func (cg *ConfigGenerator) MergeTLSConfigWithScrapeClass(tlsConfig *monitoringv1.TLSConfig, scrapeClassName string) *monitoringv1.TLSConfig {
    if scrapeClassName == "" {
        scrapeClassName = cg.defaultScrapeClassName
    }
    
    scrapeClass, found := cg.scrapeClasses[scrapeClassName]
    if !found {
        return tlsConfig
    }
    
    ...
```